autogen/retrieve_utils.py

```diff
@@ -119,15 +137,51 @@ def split_text_to_chunks(
     return chunks
 
 
+def extract_text_from_pdf(file: str) -> str:
+    """Extract text from PDF files"""
+    text = ""
+    with open(file, "rb") as f:
+        reader = pypdf.PdfReader(f)
+        if reader.is_encrypted:  # Check if the PDF is encrypted
+            try:
+                reader.decrypt("")
+            except Exception as e:
```

Comment:
Is it possible to use a specific exception?

Reply:
Done (see here for details: https://github.com/py-pdf/pypdf/blob/30db3efce36430bdbfd3412f904693c03b03daa0/pypdf/errors.py#L46C1-L47C1)